packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -385,9 +394,13 @@ export default class Editor {
             contentChanged = true;
         }
 
-        // Convert content even if it hasn't changed.
-        if (this.core.inDarkMode) {
-            const darkModeOptions = this.getDarkModeOptions();
+        const darkModeOptions = this.getDarkModeOptions();
+        // Convert when...
+        // It's in dark mode, AND it's either not the first run OR transformOnInitialze is true
+        if (
+            this.core.inDarkMode &&
+            (!firstRun || (darkModeOptions && darkModeOptions.transformOnInitialize))

```

Comment:
In what case we don't to color transformation when first run?

Reply:
If it's first run AND the setting exists AND it's set to false.

I should refactor this conditional.